src/tigerbeetle.zig

```diff
@@ -9,63 +9,62 @@ pub const config = @import("config.zig");
 
 pub const Account = packed struct {
     id: u128,
-    custom: u128,
+    /// Opaque third-party identifier to link this account (many-to-one) to an external entity:
+    user_data: u128,
+    /// Reserved for accounting policy primitives:
+    reserved: [48]u8,
+    unit: u16,
+    /// A chart of accounts code describing the type of account (e.g. clearing, settlement):
+    code: u16,
     flags: AccountFlags,
-    unit: u64,
-    debit_reserved: u64,
-    debit_accepted: u64,
-    credit_reserved: u64,
-    credit_accepted: u64,
-    debit_reserved_limit: u64,
-    debit_accepted_limit: u64,
-    credit_reserved_limit: u64,
-    credit_accepted_limit: u64,
-    padding: u64 = 0,
+    debits_reserved: u64,
+    debits_accepted: u64,
+    credits_reserved: u64,
+    credits_accepted: u64,
     timestamp: u64 = 0,
 
-    pub fn exceeds(balance: u64, amount: u64, limit: u64) bool {
-        return limit > 0 and balance + amount > limit;
+    comptime {
+        assert(@sizeOf(Account) == 128);
     }
 
-    pub fn exceeds_debit_reserved_limit(self: *const Account, amount: u64) bool {
-        return Account.exceeds(self.debit_reserved, amount, self.debit_reserved_limit);
+    pub fn debits_exceed_credits(self: *const Account, amount: u64) bool {
+        return (self.flags.debits_must_not_exceed_credits and
+            self.debits_reserved + self.debits_accepted + amount > self.credits_accepted);
```

Comment:
@adrianhopebailie this is how we calculate the new limits according to `reserved` and `accepted` balances. For example, for `debits_must_not_exceed_credits` we take the worst-case highest outcome for the debit side, i.e. `self.debits_reserved + self.debits_accepted` and ensure that this plus the amount will not exceed actual `credits_accepted` (without considering `credits_reserved` since these are as yet contingent).